CPP_STYLE_GUIDE.md

```diff
@@ -25,14 +26,29 @@
   * [Avoid throwing JavaScript errors in C++ methods](#avoid-throwing-javascript-errors-in-c)
     * [Avoid throwing JavaScript errors in nested C++ methods](#avoid-throwing-javascript-errors-in-nested-c-methods)
 
-Unfortunately, the C++ linter (based on
-[Google’s `cpplint`](https://github.com/google/styleguide)), which can be run
-explicitly via `make lint-cpp`, does not currently catch a lot of rules that are
-specific to the Node.js C++ code base. This document explains the most common of
-these rules:
+
+## Guides and References
+
+The Node.js C++ codebase strives to be consistent in its use of language
+features and idioms, as well as have some specific guidelines for the use of
+runtime features.
+
+Coding guidelines are based of following guides (highest priority first):
+1. This document
+2. The [Google C++ Style Guide][](GCSG)
+3. The ISO [C++ Core Guidelines][](CPPCG)
+
+In general code should follow the CPPCG, unless overridden by the GCSG or this
+document. At the moment these guidelines are checked manually by reviewers, with
+the goal to validate this with automatic tools.
 
 ## Formatting
 
+Unfortunately, the C++ linter (based on [Google’s `cpplint`][]), which can be
+run explicitly via `make lint-cpp`, does not currently catch a lot of rules that
+are specific to the Node.js C++ code base. This document explains the most
+common of these rules:
```

Comment:
If you want, you can remove this paragraph or merge it into the previous one, since it shares a lot of substance with it.

Reply:
Eventually, I decided to keep it as is. I think it's better to keep semantic from formatting separate, even in this case.